### ISSUE TITLE
a single request test did not accelerate

### ISSUE THREAD

**Issue by dingjingzhen** [Zefan-Cai/R-KV#8]
import json
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer
from rkv.monkeypatch import replace_llama

prompt_template = "You are given a math problem.\n\nProblem: {question}\n\n You need to solve the problem step by step. First, you need to provide the chain-of-thought, then provide the final answer.\n\n Provide the final answer in the format: Final answer:  \\boxed{{}}"

with open("./data/gsm8k.jsonl", "r") as f:
    test_data = [json.loads(line) for line in f]

sample_id = 1281
sample = test_data[sample_id]
prompt = prompt_template.format(question=sample["question"])
answer = sample["answer"].split("####")[-1].strip()
model_name = "deepseek-ai/DeepSeek-R1-Distill-Llama-8B"
compression_config = {
    "method": "rkv",
    "method_config": {
        "budget": 128,
        "window_size": 8,
        "mix_lambda": 0.07,
        "retain_ratio": 0.2,
        "retain_direction": "last",
        "record_kept_token_indices": False,
    },
    "compression": None,
    "update_kv": True
}

model_config = {
    "divide_method": "newline",
    "divide_length": 128,
    "compression_content": "all",
}
tokenizer = AutoTokenizer.from_pretrained(
    model_name, use_fast=True, padding_side="left"
)

if tokenizer.pad_token is None:
    tokenizer.pad_token = tokenizer.eos_token
    tokenizer.pad_token_id = tokenizer.eos_token_id

replace_llama(compression_config)
model = AutoModelForCausalLM.from_pretrained(
    model_name,
    torch_dtype=torch.bfloat16,
).eval()

model.config.update(model_config)

model.newline_token_ids = [
    tokenizer.encode("\n")[-1],
    tokenizer.encode(".\n")[-1],
    tokenizer.encode(")\n")[-1],
    tokenizer.encode("\n\n")[-1],
    tokenizer.encode(".\n\n")[-1],
    tokenizer.encode(")\n\n")[-1],
]

model.after_think_token_ids = [
    tokenizer.encode("</think>")[-1],
]

device = torch.device("cuda:1")
model.to(device)

inputs = tokenizer(prompt, return_tensors="pt").to(device)

import  time
for i in range(2):
    torch.cuda.synchronize()
    time1 = time.time()
    outputs = model.generate(
        **inputs,
        max_length=8192,
        num_beams=1,
        do_sample=False,
    )
    torch.cuda.synchronize()
    time2 = time.time()
    print("time:",time2 - time1)

print("Sample ID: ", sample_id)
print("Generation length:", len(outputs[0]) - inputs.input_ids.shape[1])

this is my test code,copy from analysis.ipynb,I attempted to use 'replace_llama(compression_config)' And 'model.config.update(model_config)'  These two lines of comments and uncomments are used for testing to determine whether compress is enabled.

result:
no compress:
<img width="462" height="92" alt="Image" src="https://github.com/user-attachments/assets/3c4ced8a-d460-4704-b72d-16b301f78ad4" />

compressed:
<img width="438" height="92" alt="Image" src="https://github.com/user-attachments/assets/b36e7378-a346-42bb-a7db-79d83e759b8a" />

The output after compression is longer and takes more time. However, even if the time consumption is calculated based on the generation ratio, the time consumption after compression does not have an advantage.Is there something wrong with my Settings？

**Comment by dingjingzhen** [comment]
I carefully analyzed the code and found that the bottleneck lies in "Enable Query Cache". This shouldn't be reached for every token, right? It should only be the last 8 tokens out of 128, right?

**Comment by Zefan-Cai** [comment]
There will be difference for sampling with and without KV eviction. Could you try using more budget? In a budget that is able to achieve a decent performance, the length would not be too long.

**Comment by Zefan-Cai** [comment]
> I carefully analyzed the code and found that the bottleneck lies in "Enable Query Cache". This shouldn't be reached for every token, right? It should only be the last 8 tokens out of 128, right?

Yes, only query the last local window tokens.

**Comment by dingjingzhen** [comment]
Can the hyperparameters used in gsm8k, MATH-500 and AIME 2024 be provided?

<img width="968" height="568" alt="Image" src="https://github.com/user-attachments/assets/a5f0bf5b-6a3f-4433-a008-5391bfc292a4" />
